src/backend/tools/index.ts

```diff
@@ -835,3 +840,182 @@ export async function runWineCommandOnGame(
     startFolder
   })
 }
+
+/**
+ * Download and Install the Windows version of Steam on a Wine Prefix on macOS only
+ */
+export const SteamWindows = {
+  downloadSteam: async () => {
+    const steamURL =
+      'https://drive.usercontent.google.com/download?id=1MXowtPBihbljXFs_5cM1AfyOjnC3raaT&export=download&authuser=0&confirm=t&uuid=e159346a-5836-4f7d-8aed-af3d8340fef7&at=AENtkXa2d83g4dE_UZYQKKLoEUje%3A1732794843031'
+    const directory = `${toolsPath}/steam`
+    const fileName = 'Steam.zip'
+    const window = getMainWindow()
+
+    if (!existsSync(directory)) {
+      mkdirSync(directory, { recursive: true })
+    }
+
+    if (!isOnline() || existsSync(join(directory, fileName)) || !window) {
+      return
+    }
+
+    const abortController = createAbortController('steam-download')
+
+    function handleProgress(
+      downloadedBytes: number,
+      downloadSpeed: number,
+      diskWriteSpeed: number,
+      progress: number
+    ) {
+      const currentProgress = calculateProgress(
+        downloadedBytes,
+        Number.parseInt('718394478'),
+        downloadSpeed,
+        diskWriteSpeed,
+        progress
+      )
+
+      window?.webContents.send(`progressUpdate-steam`, {
+        appName: 'steam',
+        status: 'installing',
+        runner: 'sideload',
+        progress: {
+          ...currentProgress
+        }
+      })
+    }
+
+    try {
+      await downloadFile({
+        url: steamURL,
+        dest: join(directory, fileName),
+        abortSignal: abortController.signal,
+        progressCallback: handleProgress,
+        ignoreFailure: false
+      })
+    } catch (error) {
+      logError(['Error downloading Steam', error], LogPrefix.Backend)
+      throw new Error('Error downloading Steam')
+    }
+  },
+  installSteam: async () => {
+    const steamCoverArt =
+      'https://cdn2.steamgriddb.com/file/sgdb-cdn/grid/a7e8ba67562ea4d4ca0421066466ece4.png'
+    const steamSetupPath = `${toolsPath}/steam/Steam.zip`
+    const { defaultWinePrefix, wineVersion } = GlobalConfig.get().getSettings()
+    // won't use just Steam here to avoid issue with people that already has a prefix with this name
+    const winePrefix = join(dirname(defaultWinePrefix), 'SteamHeroic')
+
+    if (!existsSync(steamSetupPath)) {
+      await SteamWindows.downloadSteam()
+    }
+
+    const gameSettings = await gameManagerMap['sideload'].getSettings('steam')
+    if (!gameSettings) {
+      return
+    }
+    writeConfig('steam', {
+      ...gameSettings,
+      winePrefix,
+      wineVersion,
+      launcherArgs:
+        '-noverifyfiles -nobootstrapupdate -skipinitialbootstrap -norepairfiles -overridepackageurl'
+    })
+
+    try {
+      await runWineCommand({
```

Comment:
would this be a problem if the default wine is crossover? we don't want to run wineboot -init in those cases, we won't have a prefix

Reply:
should we just ignore the command args in this case? I thought runWineCommand already dealt with crossover.

Reply:
I will just ignore this command then, I think I understood what you meant